From 7a0f1b78d6b8227232aadca098ba726542b0bf1f Mon Sep 17 00:00:00 2001
From: Kaviarasu Sanjeevi <quic_kaviaras@quicinc.com>
Date: Sun, 8 Sep 2024 13:58:43 +0530
Subject: [PATCH] ath11k: Compilation warining fixes for ath11k Platform files

qmi.h  -  Modified the data type of addr to u32 * in m3_dump_data structure
hw.c   -  Updated proper struct hal_rx_mpdu_info_qcn9074 in place of hal_rx_mpdu_info_qcn9074
          Modified function param to struct hal_rx_mpdu_info* instead of u8*
core.c -  Modified the function param of enable_smart_antenna from int to bool
core.h -  Modified the data type of adfs_chain_mask to unsigned long in ath11k_pdev_cap structure

Signed-off-by: Kaviarasu Sanjeevi <quic_kaviaras@quicinc.com>
---
 drivers/net/wireless/ath/ath11k/core.c | 2 +-
 drivers/net/wireless/ath/ath11k/hw.c   | 6 +++---
 drivers/net/wireless/ath/ath11k/qmi.h  | 2 +-
 drivers/net/wireless/ath/ath11k/core.h | 2 +-
 4 files changed, 6 insertions(+), 6 deletions(-)

--- a/drivers/net/wireless/ath/ath11k/core.c
+++ b/drivers/net/wireless/ath/ath11k/core.c
@@ -71,7 +71,7 @@ MODULE_PARM_DESC(ath11k_caldata_bin_path
 		"Caldata bin path for loading");
 
 bool ath11k_enable_smart_antenna = false;
-module_param_named(enable_smart_antenna, ath11k_enable_smart_antenna, uint,
+module_param_named(enable_smart_antenna, ath11k_enable_smart_antenna, bool,
 		   0644);
 MODULE_PARM_DESC(enable_smart_antenna, "Enable smart antenna supprot in fw");
 EXPORT_SYMBOL(ath11k_enable_smart_antenna);
--- a/drivers/net/wireless/ath/ath11k/hw.c
+++ b/drivers/net/wireless/ath/ath11k/hw.c
@@ -1043,7 +1043,7 @@ static void ath11k_hw_ipq5018_reo_setup(
 }
 
 static u16
-ath11k_hw_ipq8074_rx_desc_get_hal_mpdu_ppdu_id(u8 *tlv_data)
+ath11k_hw_ipq8074_rx_desc_get_hal_mpdu_ppdu_id(struct hal_rx_mpdu_info *tlv_data)
 {
 	struct hal_rx_mpdu_info *mpdu_info =
 		(struct hal_rx_mpdu_info *)tlv_data;
@@ -1053,10 +1053,10 @@ ath11k_hw_ipq8074_rx_desc_get_hal_mpdu_p
 }
 
 static
-u16 ath11k_hw_qcn9074_rx_desc_get_hal_mpdu_ppdu_id(u8 *tlv_data)
+u16 ath11k_hw_qcn9074_rx_desc_get_hal_mpdu_ppdu_id(struct hal_rx_mpdu_info *tlv_data)
 {
 	struct hal_rx_mpdu_info_qcn9074 *mpdu_info =
-		(struct hal_rx_mpdu_info_ipq9074 *)tlv_data;
+		(struct hal_rx_mpdu_info_qcn9074 *)tlv_data;
 
 	return FIELD_GET(HAL_RX_MPDU_INFO_INFO0_PPDU_ID,
 			 __le32_to_cpu(mpdu_info->info0));
--- a/drivers/net/wireless/ath/ath11k/core.h
+++ b/drivers/net/wireless/ath/ath11k/core.h
@@ -1111,7 +1111,7 @@ struct ath11k_pdev_cap {
 	u32 rx_chain_mask;
 	u32 tx_chain_mask_shift;
 	u32 rx_chain_mask_shift;
-	u32 adfs_chain_mask;
+	unsigned long adfs_chain_mask;
 	struct ath11k_band_cap band[NUM_NL80211_BANDS];
 	bool nss_ratio_enabled;
 	u8 nss_ratio_info;
--- a/drivers/net/wireless/ath/ath11k/qmi.h
+++ b/drivers/net/wireless/ath/ath11k/qmi.h
@@ -150,7 +150,7 @@ struct ath11k_qmi_m3_dump_data {
 	u32 pdev_id;
 	u32 size;
 	u64 timestamp;
-	char *addr;
+	u32 *addr;
 };
 
 struct ath11k_qmi_ce_cfg {
